src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/pack.py

```diff
@@ -0,0 +1,100 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import uuid
+import tempfile
+from knack.log import get_logger
+from knack.util import CLIError
+from azure.cli.core.commands import LongRunningOperation
+
+from ._constants import ACR_TASK_YAML_DEFAULT_NAME, ORYX_PACK_BUILDER_IMAGE
+from ._stream_utils import stream_logs
+from ._utils import (
+    validate_managed_registry,
+    get_validate_platform,
+    get_custom_registry_credentials,
+    get_yaml_and_values
+)
+from ._client_factory import cf_acr_registries
+from ._archive_utils import upload_source_code, check_remote_source_code
+from .run import prepare_source_location
+
+PACK_NOT_SUPPORTED = 'Pack is only available for managed registries.'
+PACK_TASK_YAML_FMT = '''steps:
+  - cmd: mcr.microsoft.com/oryx/pack:stable build {image_name} --builder {builder} --env REGISTRY_NAME={{{{.Run.Registry}}}} -p .
+  - push: ["{image_name}"]
+'''
+
+logger = get_logger(__name__)
+
+
+def acr_pack(cmd,  # pylint: disable=too-many-locals
+             client,
+             registry_name,
+             image_name,
+             source_location,
+             builder=ORYX_PACK_BUILDER_IMAGE,
+             no_format=False,
+             no_logs=False,
+             no_wait=False,
+             timeout=None,
+             resource_group_name=None,
+             os_type=None,
```

Comment:
We have removed the deprecated `os_type` parameter and incorporated it as a `--platform` option. See the change here:
https://github.com/Azure/azure-cli/pull/9317/commits/d3d97218f4a2df2276ee1f193ff6a3023eac8177